distbelief/optim/downpour_sgd.py

```diff
@@ -44,14 +40,9 @@ def __init__(self, params, lr=required, freq=required, model=required, internal_
         send_message(MessageCode.ParameterUpdate, ravel_model_params(self.model))
         self.idx = 0
 
-        # create a listener that runs in a separate thread, checking for info from servers.
         listener = DownpourListener(self.model)
-        print('starting thread')
-        t = Thread(target=listener.run)
-        t.start()
+        listener.start()
```

Comment:
Will this eventually call `listener.run()`? Even if the `DownpourListener` is a `Thread`, calling the superclass `start()` method won't result in a call to `listener.run()`.

Reply:
Yes, I beleive so it seems like this is the style use here:
https://stackoverflow.com/questions/660961/overriding-python-threading-thread-run